builders/ethereum/dev-env/thirdweb.md

```diff
@@ -110,184 +112,339 @@ thirdweb offers SDKs for a range of programming languages, such as React, React
+
+This guide will show you how to interact with your contracts deployed to Moonbeam, Moonriver or Moonbase Alpha via thirdweb using Typescript. You can view the [full SDK reference](https://portal.thirdweb.com/typescript/v5){target=\_blank} in thirdweb’s documentation.
+
+### Run Locally {: #run-locally }
+
+To run your dApp locally for testing and debugging purposes, use the command: 
 
 ```bash
-npx thirdweb create app --evm
+yarn dev
 ```
 
-Or install it into your existing project by running:
+The app will compile and specify the localhost and port number for you to visit in your browser.
 
-```bash
-npx thirdweb install
+![thirdweb run locally](/images/builders/ethereum/dev-env/thirdweb/thirdweb-4.webp)
+
+### Create Client ID {: #create-client-id }
+
+If you generated your thirdweb app with Vite, you'll have a `client.ts` file that looks like the below. As long you've created a `.env` file with your thirdweb API Key (Client ID) defined in `VITE_TEMPLATE_CLIENT_ID`, you can leave the `client.ts` as is and proceed to the next section.
+
+```typescript title="client.ts"
+import { createThirdwebClient } from 'thirdweb';
+
+// Replace this with your client ID string
+// refer to https://portal.thirdweb.com/typescript/v5/client on how to get a client ID
+const clientId = import.meta.env.VITE_TEMPLATE_CLIENT_ID;
+
+export const client = createThirdwebClient({
+  clientId: clientId,
+});
 ```
 
-### Initialize SDK On Moonbeam {: #initialize-sdk-on-moonbeam }
+### Configure Chain {: #configure-chain }
 
-Wrap your application in the `ThirdwebProvider` component and change the `activeChain` to Moonbeam.
+thirdweb offers a small number of chains from `@thirdweb/chains` and does not include Moonbeam networks in that list, so you'll need to specify the network details including chain ID and RPC URL. You can create a custom chain with [`defineChain`](https://portal.thirdweb.com/references/typescript/v5/defineChain){target=\_blank} as follows:
 
-```javascript
-import { ThirdwebProvider } from '@thirdweb-dev/react';
-import { Moonbeam } from '@thirdweb-dev/chains';
+=== "Moonbeam"
 
-const App = () => {
-  return (
-    <ThirdwebProvider activeChain={Moonbeam}>
-      <YourApp />
-    </ThirdwebProvider>
-  );
-};
+    ```typescript
+    import { defineChain } from 'thirdweb';
+    const moonbase = defineChain({
+    id: {{ networks.moonbeam.chain_id }},
+    rpc: '{{ networks.moonbeam.public_rpc_url }}',
+    })
+    ```
+
+=== "Moonriver"
+
+    ```typescript
+    import { defineChain } from 'thirdweb';
+    const moonbase = defineChain({
+    id: {{ networks.moonriver.chain_id }},
+    rpc: '{{ networks.moonriver.public_rpc_url }}',
+    })
+    ```
+
+=== "Moonbase Alpha"
+
+    ```typescript
+    import { defineChain } from 'thirdweb';
+    const moonbase = defineChain({
+    id: {{ networks.moonbase.chain_id }},
+    rpc: '{{ networks.moonbase.rpc_url }}',
+    })
+    ```
```

Comment:
The code here needs to be formatted. I still think the title of the file these snippets get added to should be defined